Dfe.Academies.External.Web/Pages/WhatAreYouApplyingToDo.cshtml.cs

```diff
@@ -1,83 +1,77 @@
 using Dfe.Academies.External.Web.Attributes;
 using Dfe.Academies.External.Web.Enums;
 using Dfe.Academies.External.Web.Models;
+using Dfe.Academies.External.Web.Pages.Base;
 using Dfe.Academies.External.Web.Services;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.AspNetCore.Mvc.RazorPages;
 
-namespace Dfe.Academies.External.Web.Pages
+namespace Dfe.Academies.External.Web.Pages;
+
+public class WhatAreYouApplyingToDoModel : BasePageModel
 {
-    public class WhatAreYouApplyingToDoModel : PageModel
+    private readonly ILogger<WhatAreYouApplyingToDoModel> _logger;
+    private readonly IConversionApplicationCreationService _academisationCreationService;
+    private const string NextStepPage = "/WhatIsYourRole";
+
+    public WhatAreYouApplyingToDoModel(ILogger<WhatAreYouApplyingToDoModel> logger, IConversionApplicationCreationService academisationCreationService)
     {
-        private readonly ILogger<WhatAreYouApplyingToDoModel> _logger;
-        private readonly IConversionApplicationCreationService _academisationCreationService;
-        private const string NextStepPage = "/WhatIsYourRole";
+        _logger = logger;
+        _academisationCreationService = academisationCreationService;
+    }
 
-        public WhatAreYouApplyingToDoModel(ILogger<WhatAreYouApplyingToDoModel> logger, IConversionApplicationCreationService academisationCreationService)
-        {
-            _logger = logger;
-            _academisationCreationService = academisationCreationService;
-        }
+    [BindProperty]
+    [RequiredEnum(ErrorMessage = "Select an application type")]
+    public ApplicationTypes ApplicationType { get; set; }
 
-        [BindProperty]
-        [RequiredEnum(ErrorMessage = "Select an application type")]
-        public ApplicationTypes ApplicationType { get; set; }
+    public async Task OnGetAsync()
+    {
+        // like on load - if navigating backwards from NextStepPage - will need to set model value from somewhere!
+        // ApplicationType = TempData["applicationTypeSelected"];
+    }
 
-        public async Task OnGetAsync()
+    public async Task<IActionResult> OnPostAsync()
+    {
+        if (!ModelState.IsValid)
         {
-            // like on load - if navigating backwards from NextStepPage - will need to set model value from somewhere!
-            // ApplicationType = TempData["applicationTypeSelected"];
+            // error messages component consumes ViewData["Errors"]
+            ViewData["Errors"] = ConvertModelDictionary();
```

Comment:
Rather than passing this down to the PartialView in the ViewData for the page, I'd rather bind this as the Model behind the PartialView, otherwise the PartialView has to make assumptions about what the page is providing for it.

Reply:
`<partial name="_ValidationSummary" model=@Model.ConvertModelDictionary()/>`

Reply:
agreed, I've done this in next PR

Reply:
changed my mind, done that in here in this branch now